upf/upf_pfcp_server.c

```diff
@@ -1212,7 +1212,20 @@ static uword
 						      (upf_event_urr_hdr_t));
 		upf_session_t *sx;
 
+		/*
+		 * The session could have been freed or replaced after
+		 * this even has been queued, but before it has been
+		 * handled. So we check if the pool entry is free, and
+		 * also verify that CP-SEID in the session matches
+		 * CP-SEID in the event.
+		 */
+		if (pool_is_free_index (gtm->sessions, ueh->session_idx))
+		  continue;
```

Comment:
The continue skips over the `vec_free_h` at the end. Wouldn't this cause a leak of the ueh entry?

Reply:
Thanks for noticing this problem. Adjusted the code slightly to avoid the leak